### PR TITLE
Chapter 8: Basic login

### PR DESCRIPTION
**Pull request by nguyenthanhtu95**

### CODE REVIEW

**Comment by nguyenthanhtu95** [comment]
ready

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  def new; end
+
+  def create
+    user = User.find_by(email: params[:session][:email].downcase)
+    if user && user.authenticate(params[:session][:password])
+      log_in user
+      redirect_to user
+    else
+      flash.now[:danger] = "Invalid email/password combination"
+      render "new"
```
Comment (phamvanbk11) [inline app/controllers/sessions_controller.rb]: :new

File: app/controllers/sessions_controller.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+class SessionsController < ApplicationController
+  def new; end
+
+  def create
+    user = User.find_by(email: params[:session][:email].downcase)
+    if user && user.authenticate(params[:session][:password])
+      log_in user
+      redirect_to user
+    else
+      flash.now[:danger] = "Invalid email/password combination"
```
Comment (phamvanbk11) [inline app/controllers/sessions_controller.rb]: I18n nhé

File: app/views/layouts/_header.html.erb [hunk 3]
```diff
@@ -3,9 +3,26 @@
...
+        <li><%= link_to "Help", help_path %></li>
+        <% if logged_in? %>
+          <li><%= link_to "Users", '#' %></li>
+          <li class="dropdown">
+            <a href="#" class="dropdown-toggle" data-toggle="dropdown">
+              Account <b class="caret"></b>
+            </a>
+            <ul class="dropdown-menu">
+              <li><%= link_to "Profile", current_user %></li>
+              <li><%= link_to "Settings", '#' %></li>
```
Comment (phamvanbk11) [inline app/views/layouts/_header.html.erb]: I18n nhé

**Comment by phamvanbk11** [comment]
commented

File: app/controllers/sessions_controller.rb (hunk 1 above)
Comment (nguyenthanhtu95) [reply app/controllers/sessions_controller.rb]: fixed!

File: app/controllers/sessions_controller.rb (hunk 2 above)
Comment (nguyenthanhtu95) [reply app/controllers/sessions_controller.rb]: fixed!

File: app/views/layouts/_header.html.erb (hunk 3 above)
Comment (nguyenthanhtu95) [reply app/views/layouts/_header.html.erb]: fixed!

**Comment by nguyenthanhtu95** [comment]
ready

**Comment by phamvanbk11** [comment]
merged

**Pull request merged**